azurelinuxagent/common/protocol/wire.py

```diff
@@ -781,76 +782,67 @@ def update_goal_state(self, force_update=False):
         Updates the goal state if the incarnation or etag changed or if 'force_update' is True
         """
         try:
-            updated = False
-
             goal_state = GoalState(self)
 
             # Always update the hostgaplugin, since the agent may issue requests to it even if there are no other changes
             # in the goal state (e.g. report status)
             self._update_host_plugin(goal_state.container_id, goal_state.role_config_name)
 
+            # Fetch the full goal state if needed
```

Comment:
Not sure if this comment is  appropriate here. It threw me a little off which trying to understand the code. Guess the right place is at line 795 after the if statement and before fetch_full_goal_state?

Reply:
i can remove the comment altogether if you find it confusing, let me know